frontend/src/pages/create_campaign/index.js

```diff
@@ -1,34 +1,76 @@
-import {
-  FormControlLabel,
-  Switch,
-} from "@mui/material";
-import AdapterDateFns from '@mui/lab/AdapterDateFns';
-import LocalizationProvider from '@mui/lab/LocalizationProvider';
-import DateTimePicker from '@mui/lab/DateTimePicker';
-import React from "react";
-import { useNavigate } from "react-router-dom";
-import { CampaignContainer, CampaignDropzone, CampaignRowDiv, CampaignSubmit, CampaignTextField } from "./createCampaign.styled";
-import Dropzone from 'react-dropzone';
-import { LoadingIndicator } from "../../components";
-import { fileToDataUrl } from "../../utils";
+import React, { useState } from "react";
+import { Container, Tabs, Tab } from "@mui/material";
+import CampaignTab from "./Campaign";
+import RolesTab from "./Roles";
+import ReviewTab from "./Preview";
+import { NextButton, ArrowIcon, NextWrapper } from "./createCampaign.styled";
 
+// FIXME: CHAOS-66, user authentication and redirection if they are not logged in or authenticated
 const CreateCampaign = () => {
-  const navigate = useNavigate();
-
-  const [name, setName] = React.useState("");
-  const [startDate, setStartDate] = React.useState(new Date());
-  const [endDate, setEndDate] = React.useState(new Date());
-  const [description, setDescription] = React.useState("");
-  const [interviewStage, setInterviewStage] = React.useState(false);
-  const [scoringState, setScoringStage] = React.useState(false);
-  const [draft, setDraft] = React.useState(false);
-  const [cover, setCover] = React.useState(null);
-  const [error, setError] = React.useState(null);
+  const [tab, setTab] = useState(0);
+  const [campaignName, setCampaignName] = useState("");
+  const [startDate, setStartDate] = useState(new Date());
+  const [endDate, setEndDate] = useState(new Date());
+  const [description, setDescription] = useState("");
+  const [interviewStage, setInterviewStage] = useState(false);
+  const [scoringStage, setScoringStage] = useState(false);
+  const [cover, setCover] = useState(null);
+  const [error, setError] = useState(null);
+  const [roles, setRoles] = useState([]);
+  const [roleSelected, setRoleSelected] = useState(
+    roles.length > 0 ? roles[0].id : "-1"
+  );
+  const [questions, setQuestions] = useState([]);
+  const campaignData = {
+    tab,
+    setTab,
+    campaignName,
+    setCampaignName,
+    startDate,
+    setStartDate,
+    endDate,
+    setEndDate,
+    description,
+    setDescription,
+    interviewStage,
+    setInterviewStage,
+    scoringStage,
+    setScoringStage,
+    cover,
+    setCover,
+    error,
+    setError,
+    roles,
+    setRoles,
+    roleSelected,
+    setRoleSelected,
+    questions,
+    setQuestions,
+  };
 
-  // TODO: user authentication and redirection if they are not logged in or authenticated
+  const onTabChange = (val) => {
+    // only allow user to access review/publish tab if all inputs are non-empty
+    if (val === 2) {
```

Comment:
what is val === 2? not quite clear. i understand this is trying to check all fields are non empty but not sure what this clause does